### ISSUE TITLE
Add support for setting a Personal Identity Number from code

### ISSUE THREAD

**Issue by fredrik-lundin** [ActiveLogin/ActiveLogin.Authentication#17]
The Grand ID API (Svensk e-identitet) supports adding `pnr` (short for _personnummer (Swedish personal number))_ property to authentication request to them. 

When the `pnr` property is included in the request, the client will not have to fill out the Swedish personal number from Svensk e-identitets UI, but can choose to use a customized form for that on its own site or app.

**Comment by PeterOrneholm** [comment]
Great idea!

I'd propose we call the property "swedishPersonalIdentityNumber" and you pass it as a property when doing the callange, like this:

```csharp
var props = new AuthenticationProperties
{
    RedirectUri = Url.Action(nameof(ExternalLoginCallback)),
    Items =
    {
        {"returnUrl", returnUrl},
        {"scheme", provider},
        {"swedishPersonalIdentityNumber", "XXXXXX-YYYY"}
    }
};

return Challenge(props, provider);
```
Would that work? This should be implemented for both `.AddBankID()` and `.AddGrandId()`.
I'd like to be more specific than pnr (even if GrandID uses that as query param) and use `swedishPersonalIdentityNumber` instead.

**Comment by PeterOrneholm** [comment]
I updated the title to reflect that, if implemented, it should be done for both GrandID and BankID.

**Comment by fredrik-lundin** [comment]
Sending it via the items property during a challenge would be my preferred approach too. So that looks good to me :)

I'd say `swedishPersonalIdentityNumber` is fine as a name. Let's implement this!

**Comment by nikolaykrondev** [comment]
I would like to not have that value as hardcoded directly, what do you think if we extract it as 
```csharp
    public class AuthenticationConstants
    {
        public class Items
        {
            public const string SwedishPersonalIdentityNumber = "swedishPersonalIdentityNumber";
        }
    }
```
under Common? So the easy usage would be:
```csharp
     var props = new AuthenticationProperties
            {
                RedirectUri = Url.Action(nameof(ExternalLoginCallback)),
                Items =
                {
                    {"returnUrl", returnUrl},
                    {"scheme", provider},
                    {AuthenticationPropertiesConstants.Items.SwedishPersonalIdentityNumber, "661105-3718"}
                }
            };
```

**Comment by PeterOrneholm** [comment]
> I would like to not have that value as hardcoded directly, what do you think if we extract it as

I really like the idea of a constant, but I'd like to not put that in Common. The things in common should never be consumed by anything else than our own packages I'd say.

We should probably add it to `BankIdAuthenticationConstants` and `GrandIdAuthenticationConstants` and maybe call it `AuthenticationPropertyItemSwedishPersonalIdentityNumber`.

Does @fredrik-lundin have any input on this?

**Comment by fredrik-lundin** [comment]
> > I would like to not have that value as hardcoded directly, what do you think if we extract it as
> 
> I really like the idea of a constant, but I'd like to not put that in Common. The things in common should never be consumed by anything else than our own packages I'd say.
> 
> We should probably add it to `BankIdAuthenticationConstants` and `GrandIdAuthenticationConstants` and maybe call it `AuthenticationPropertyItemSwedishPersonalIdentityNumber`.
> 
> Does @fredrik-lundin have any input on this?

Sorry for late reply! This looks fine to me :)

**Comment by PeterOrneholm** [comment]
Fixed in #30
